apps/antidote/src/meta_data_notif_server.erl

```diff
@@ -0,0 +1,336 @@
+%% @doc Meta data notification server. Allows client code to subscribe to the
+%% event when the provided snapshot becomes stable.
+
+-module(meta_data_notif_server).
+
+-include("antidote.hrl").
+-include("meta_data_notif_server.hrl").
+
+-ifdef(EUNIT).
+-include_lib("proper/include/proper.hrl").
+-include_lib("eunit/include/eunit.hrl").
+-endif.
+
+-export([start_link/0, start/0,
+         init/0,
+         loop/1,
+         subscribe_to_stable/1,
+         notify/2
+        ]).
+
+-record(state, { mons = [] :: [{pid(), reference()}],
+                 %% NOTE: vector_orddict stores Big -> Small
+                 snaps = vector_orddict:new() ::
+                           vector_orddict:vector_orddict([pid()]),
+                 min_snapshot :: undefined | antidote:snapshot_time()
+               }).
+-type state() :: #state{}.
+
+start_link() ->
+    proc_lib:start_link(?MODULE, init, []).
+
+start() ->
+    proc_lib:start(?MODULE, init, []).
+
+%% @doc API to subscribe to stable snapshot notifications. Notification server
+%% will send notification of the form `#md_stable_snapshot{}` to the calling process.
+%% Same client may subscribe to multiple snapshots at the same time.
+-spec subscribe_to_stable(antidote:snapshot_time()) -> ok.
+subscribe_to_stable(Snapshot) ->
+    whereis(?MODULE) ! {add_snapshot, Snapshot, self()},
+    ok.
+
+%% @doc Meta data server notifies notif server about new stable snapshot.
+-spec notify(stable_time_functions, antidote:snapshot_time()) -> ok.
+notify(stable_time_functions, Snapshot) ->
+    true = maybe_notify_about_snapshot(Snapshot),
+    whereis(?MODULE) ! {stable_snapshot, Snapshot},
+    ok.
+
+%%------------------------------------------------------------------------------
+
+init() ->
+    try erlang:register(?MODULE, self()) of
+        true ->
+            proc_lib:init_ack({ok, self()}),
+            loop(#state{})
+    catch _:_ ->
+            proc_lib:init_ack({error, already_registered})
+    end.
+
+loop(State) ->
```

Comment:
Just curious, why is this not implemented as a gen_server or other OTP process?

Reply:
Ah, got it :) - it just doesn't use call backs.

Reply:
There are various ways how this process can be implemented, and I just use proc_lib process for simplicity. I was planning to get back to it, once we consider this to be merged back to Antidote, but for now the idea was that server just acts as a simple notification server primitive. I'm mostly curious if local DC calculation that takes place in this PR makes sense or not.